code/modules/paperwork/paper.dm

```diff
@@ -298,56 +298,62 @@
 		return
 	switch(action)
 		if("stamp")
+			if(length(stamps) >= MAX_PAPER_STAMPS)
+				to_chat(usr, pick("You try to stamp but you miss!", "There is no where else you can stamp!"))
+				return FALSE
+
 			var/stamp_x = text2num(params["x"])
 			var/stamp_y = text2num(params["y"])
 			var/stamp_r = text2num(params["r"])	// rotation in degrees
-			var/stamp_icon_state = params["stamp_icon_state"]
-			var/stamp_class = params["stamp_class"]
+
+			var/obj/holding = ui.user.get_active_held_item()
+			var/real_stamp = istype(holding, /obj/item/stamp)
+			var/datum/asset/spritesheet/sheet = get_asset_datum(/datum/asset/spritesheet/simple/paper)
+			var/stamp_icon_state = real_stamp ? holding.icon_state : "FAKE"
+			var/stamp_class = real_stamp ? sheet.icon_class_name(holding.icon_state) : "FAKE"
```

Comment:
Why isn't this an early return? What does stamping with "FAKE" imply?

Reply:
`FAKE` results in using the fallback stamp icon, but there's no reason this can't just be an early return.